src/internal/connector/exchange/service_iterators.go

```diff
@@ -226,6 +243,14 @@ func FetchEventIDsFromCalendar(
 	for {
 		resp, err := builder.Get(ctx, options)
 		if err != nil {
+			if hasErrorCode(err, errItemNotFound) {
+				// The folder was deleted between the time we populated the container
+				// cache and when we tried to fetch data for it. All we can do is
+				// return no jobs because we've only pulled basic info about each
+				// item.
+				return nil, "", nil
```

Comment:
This is kinda tough, but `nil, nil` returns tend to be frowned upon.  Could we return and handle/discard `errItemNotFound` or some similar in the caller?